### PR TITLE
Create endpoints to Create, Update, Delete users

### PR DESCRIPTION
**Pull request by pattrickrice**
- Import security module for hashing passwords.
- Created create, update, delete api endpoints.
- Still need orm hookup.

### CODE REVIEW

File: src/main/java/github/paz/awardportal/controller/UserController.java [hunk 1]
```diff
@@ -57,10 +57,10 @@
             @ApiResponse(code = 500, message = "Failed to create the user. Try again later.")
     })
     public ResponseEntity<String> createUser(
-            @RequestParam("firstName") String firstName,
-            @RequestParam("lastName") String lastName,
-            @RequestParam("email") String email,
-            @RequestParam("password") String password) {
+            @RequestHeader("firstName") String firstName,
```
Comment (anderMatt) [inline src/main/java/github/paz/awardportal/controller/UserController.java]: Looks good! I'd like to point out another Spring annotation that could potentially be used here, @RequestBody. (https://www.baeldung.com/spring-request-response-body)

Suppose we POST the new user json to this endpoint:

{"firstName": "name", "lastName": "name", "email": "theEmail"...}

The @RequestBody annotation will automatically instantiate a User object from the json, and pass it to the controller method. 
 
createUser(@RequestBody User user) {   //Spring instantiates the user from the json data.
    <do stuff with the User object...hash its password, pass to database code, etc.>
    return ResponseEntity.ok();
}

This is just a suggestion that allows us to take full advantage of some utilities Spring provides, but passing data in headers and instantiating User objects ourselves is perfectly fine too!

**Review by anderMatt — APPROVED** [review]

File: src/main/java/github/paz/awardportal/controller/UserController.java (hunk 1 above)
Comment (anderMatt) [reply src/main/java/github/paz/awardportal/controller/UserController.java]: Edit: Just noticed first name is a Header and not RequestParam - was this intended or a mistake?

Comment (pattrickrice) [reply src/main/java/github/paz/awardportal/controller/UserController.java]: It was just a mistake.

**Comment by pattrickrice** [comment]
Thanks for suggesting the Requestbody annotation, I hadn't seen that yet. This new request implements that.

**Pull request merged**